### ISSUE TITLE
tags/extra info about control in wrapper profile not reflected accurately in automate reports

### ISSUE THREAD

**Issue by vjeffrey** [chef/automate#1009]
## Describe the bug
When creating a wrapper profile, tags defined for a control via the wrapper profile ~do not show up in Automate.~ show up in the Automate view of the profile details and report, but the content defined in the wrapper profile completely overwrites the original content.

## To Reproduce
- create a wrapper profile
- modify or add tags for a control via the wrapper profile 

## Expected behavior
Defined tags are all you can see about the control when viewing control source.

## Versions (please complete the following information):
- Automate Build Number: 20190722170639

**Comment by vjeffrey** [comment]
[wrapper-example-1.0.1.tar.gz](https://github.com/chef/automate/files/3419896/wrapper-example-1.0.1.tar.gz)

I created the above profile for testing.
In my attempt to reproduce this issue, I've found that when viewing both the uploaded profile and the report details, the tag information included in the wrapper *is included* but overwrites the entire control (view only - it still executes the actual control code). 

```
include_controls 'ssl-baseline' do
  control 'robotattack' do
    tag 'monster attack'
  end

  control 'mac-null' do
    tag 'monster attack'
  end
end
```
<img width="1034" alt="Screen Shot 2019-07-22 at 20 25 28" src="https://user-images.githubusercontent.com/10341541/61678245-87e25980-acbf-11e9-84d2-6aba9919c203.png">
<img width="1210" alt="Screen Shot 2019-07-22 at 20 25 43" src="https://user-images.githubusercontent.com/10341541/61678246-87e25980-acbf-11e9-8f26-ad48ff4bbf60.png">

**Comment by alexpop** [comment]
I looked into this issue. Thank you for providing the sample profile and repro steps.

Indeed, inspec is executing the updated control. Below is a snippet of an exec output where details of control `robotattack` are shown. Observe how the `monster attack` tag, added via wrapping, is showing in the control tags, but not in the `code` block:

```
$ inspec exec wrapper-example-1.0.1.tar.gz --reporter json-automate | jq

{
  "platform": {
    "name": "mac_os_x",
    "release": "17.7.0"
  },
  "profiles": [
    {
      "name": "wrapper-example",
      "version": "1.0.1",
      "sha256": "8787a3b453167c57f5b3f57f2a124054cda9113d87fd5e2e0761520937cc27f1",
      "title": "Wrapper Profile",
      "summary": "Base profile that wraps other profile",
      "supports": [],
      "attributes": [],
      "depends": [
        {
          "name": "ssl-baseline",
          "url": "https://github.com/dev-sec/ssl-baseline/archive/1.4.0.tar.gz",
          "status": "loaded"
        }
      ],
      "controls": [
...
        {
          "id": "robotattack",
          "title": "Return Of Bleichenbacher's Oracle Threat",
          "desc": "ROBOT is the return of a 19-year-old vulnerability that allows performing RSA decryption and signing operations with the private key of a TLS server.",
          "descriptions": [
            {
              "label": "default",
              "data": "ROBOT is the return of a 19-year-old vulnerability that allows performing RSA decryption and signing operations with the private key of a TLS server."
            }
          ],
          "impact": 0.5,
          "refs": [
            {
              "url": "https://ia.cr/2017/1189",
              "ref": "Paper: Return Of Bleichenbacher's Oracle Threat (ROBOT)"
            },
            {
              "ref": []
            }
          ],
          "tags": {
            "sslattack": null,
            "tlsattack": null,
            "monster attack": null
          },
          "code": "control 'robotattack' do\n  title \"Return Of Bleichenbacher's Oracle Threat\"\n  desc 'ROBOT is the return of a 19-year-old vulnerability that allows performing RSA decryption and signing operations with the private key of a TLS server.'\n  ref \"Paper: Return Of Bleichenbacher's Oracle Threat (ROBOT)\", url: 'https://ia.cr/2017/1189'\n  tag 'sslattack', 'tlsattack'\n  impact 0.5\n  only_if { sslports.length > 0 }\n\n  sslports.each do |sslport|\n    # create a description\n    proc_desc = \"on node == #{target_hostname} running #{sslport[:socket].process.inspect} (#{sslport[:socket].pid})\"\n    describe ssl(sslport).ciphers(/^TLS_RSA/i) do\n      it(proc_desc) { should_not be_enabled }\n      it { should_not be_enabled }\n    end\n  end\nend\n",
          "source_location": {
            "line": 565,
            "ref": "/Users/apop/.inspec/cache/f1e3c476bfa1711c80fa6f20bd23fa50811d36f62fed613bcb851998f1cbd087/ssl-baseline-1.4.0/controls/ssl_test.rb"
          },
          "results": [
            {
              "status": "failed",
              "code_desc": "SSL/TLS on localhost:63342 with cipher == /^TLS_RSA/i on node == mymac running \"goland\" (613)",
              "run_time": 0.030927,
              "start_time": "2019-07-26T05:48:37+01:00",
              "message": "expected SSL/TLS on localhost:63342 with cipher == /^TLS_RSA/i not to be enabled"
            },
            {
              "status": "failed",
              "code_desc": "SSL/TLS on localhost:63342 with cipher == /^TLS_RSA/i should not be enabled",
              "run_time": 0.021103,
              "start_time": "2019-07-26T05:48:37+01:00",
              "message": "expected SSL/TLS on localhost:63342 with cipher == /^TLS_RSA/i not to be enabled"
            }
          ]
        }
      ],
```

If this ^ report is ingested without the profile being uploaded to Automate beforehand, it will show a code block without the `monster attack` tag.

---

If the profile is uploaded to Automate first, we run inspec json on it to collect the profile metadata. Let's look at the output of that.

```
$ inspec json wrapper-example-1.0.1.tar.gz | jq
{
  "name": "wrapper-example",
  "title": "Wrapper Profile",
  "summary": "Base profile that wraps other profile",
  "version": "1.0.1",
  "depends": [
    {
      "name": "ssl-baseline",
      "url": "https://github.com/dev-sec/ssl-baseline/archive/1.4.0.tar.gz",
      "status": "loaded"
    }
  ],
  "supports": [],
  "controls": [
...
    {
      "title": "Return Of Bleichenbacher's Oracle Threat",
      "desc": "ROBOT is the return of a 19-year-old vulnerability that allows performing RSA decryption and signing operations with the private key of a TLS server.",
      "descriptions": {
        "default": "ROBOT is the return of a 19-year-old vulnerability that allows performing RSA decryption and signing operations with the private key of a TLS server."
      },
      "impact": 0.5,
      "refs": [
        {
          "url": "https://ia.cr/2017/1189",
          "ref": "Paper: Return Of Bleichenbacher's Oracle Threat (ROBOT)"
        },
        {
          "ref": []
        }
      ],
      "tags": {
        "sslattack": null,
        "tlsattack": null,
        "monster attack": null
      },
      "code": "  control 'robotattack' do\n    tag 'monster attack'\n  end\n",
      "source_location": {
        "ref": "/Users/apop/.inspec/cache/f1e3c476bfa1711c80fa6f20bd23fa50811d36f62fed613bcb851998f1cbd087/ssl-baseline-1.4.0/controls/ssl_test.rb",
        "line": 565
      },
      "id": "robotattack"
    }
```

^ Here we are seeing a different behavior. While control tag `monster attack` is showing up for the `robotattack` control, the `code` block is only showing the properties that have been overwritten in the wrapper profile. This leads to the bug reported in this issue.

To sum up, control code blocks are incomplete for controls updated via wrapper profiles. Applies to `inspec exec` and `inspec json` outputs.

**Comment by kekaichinose** [comment]
@alexpop and @vjeffrey sorry, I got confused on this thread. It sounds like this needs to go to InSpec team next, yeah? If so, can you move this issue over to the inspec/inspec project and I can work with the team to prioritize. Thanks!

**Comment by vjeffrey** [comment]
@alexpop will be opening an issue in inspec for this; closing this issue now

**Comment by alexpop** [comment]
@kekaichinose I opened https://github.com/inspec/inspec/issues/4395 for the `inspec exec` and `inspec json` output issues of wrapped profiles.

Closing this Automate ticket...
